src/detect-mqtt-subscribe-topic.c

```diff
@@ -153,7 +155,7 @@ static void PrefilterTxMQTTSubscribeTopic(DetectEngineThreadCtx *det_ctx,
     const int list_id = ctx->list_id;
 
     int local_id = 0;
-    while(1) {
+    while ((subscribe_topic_match_limit == 0) || local_id < subscribe_topic_match_limit) {
```

Comment:
this enforces the limit in detection, but will still parse all the topics and also still log all of them. Is that desired? I'd expect a limit like this to be enforced during parsing to avoid resource starvation during all processing.

Reply:
From my point of view, this is desired. For logging, I would always want to see everything that is going on, as I would consider this the main use case for this parser. If you disagree, we could surely also make this an option in the `app-layer.mqtt` config section.

Reply:
OK with leaving parsing as is for now? Then I will open a new PR.

Reply:
I am not sure that having many topics in one transaction is the most expensive way, ie to get resource starvation. I think it is more effective, from an adversary, to send the same packet with a single topic multiple times...
What do you think @satta ?

Reply:
I tend to agree but have not measured it.
I just wanted to say that for logging I would really _not_ expect any truncation beyond the cutoff based on raw message size.

Reply:
is there a max number of topics defined by the protocol?

Reply:
There is, and it is implicitly defined by the maximal size of a "message" and the space requirement per topic. See https://github.com/OISF/suricata/pull/5622#discussion_r628123516 -- about 67M topics max in the SUBSCRIBE case, for example.

Reply:
@victorjulien @catenacyber can we resolve this discussion or do you think I should also limit topic parsing in addition to detection?

Reply:
For me it is ok